finger-exercises/redux/src/app/screens/App/index.js

```diff
@@ -1,38 +1,37 @@
 import React, { Component, Fragment } from 'react';
-import store from '@redux/store';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { bookSelectedPropType, booksPropType } from '@constants/propTypes';
 import Navbar from '@components/Navbar';
 import Footer from '@components/Footer';
+import bookActions from '@redux/book/actions';
 
 import Book from './components/Book';
 import Search from './components/Search';
 import ShoppingCart from './components/ShoppingCart';
 import styles from './styles.scss';
 
 class App extends Component {
-  state = {
-    books: [],
-    bookSelected: []
-  };
-
   componentDidMount() {
-    store.subscribe(() => {
-      const { books, bookSelected } = store.getState();
-      this.setState({ books, bookSelected });
-    });
-    // TODO to implement the dispatch
+    this.props.getBooks();
   }
 
-  // TODO to implement the dispatch
-  onSearch = value => {};
+  onSearch = value => {
+    this.props.getBooks();
+    this.props.searchBook(value);
+  };
 
-  // TODO to implement the dispatch
-  addToCart = item => {};
+  addToCart = item => {
```

Comment:
`addToCart`, `addItem` and `removeItem` are just calling their props analog. Do we need them?